salt/minion.py

```diff
@@ -1323,35 +1372,40 @@ def _handle_decoded_payload(self, data):
         instance = self
         multiprocessing_enabled = self.opts.get('multiprocessing', True)
         if multiprocessing_enabled:
-            if sys.platform.startswith('win'):
-                # let python reconstruct the minion on the other side if we're
-                # running on windows
-                instance = None
             with default_signals(signal.SIGINT, signal.SIGTERM):
-                process = SignalHandlingMultiprocessingProcess(
-                    target=self._target, args=(instance, self.opts, data, self.connected)
-                )
+                if sys.platform.startswith('win'):
+                    # let python reconstruct the minion on the other side if we're
+                    # running on windows
+                    instance = None
+                    process = SignalHandlingMultiprocessingProcess(target=self._target,
+                                                                   args=(instance, self.opts,
+                                                                         data, self.connected, None))
+                else:
+                    # Process pooling only on Unix at the moment
+                    CallbackProcessPool().add_data(data)
```

Comment:
This seems like it's going to instantiate that pool every time when it's really only necessary in the rare case where somebody requests queuing, correct?